manifests/charts/istio-control/istio-discovery/templates/configmap.yaml

```diff
@@ -47,9 +47,12 @@
       {{- /* Fill in openCensusAgent configuration from meshConfig so it isn't overwritten below */ -}}
         {{ toYaml $.Values.meshConfig.defaultConfig.tracing }}
       {{- end }}
-
       {{- if .Values.global.remotePilotAddress }}
-      discoveryAddress: {{ printf "istiod-remote.%s.svc" .Release.Namespace }}:15012
+        {{- if .Values.pilot.enabled }}
+        discoveryAddress: {{ printf "istiod-remote.%s.svc" .Release.Namespace }}:15012
+        {{- else }}
+        discoveryAddress: {{ printf "istiod.%s.svc" .Release.Namespace }}:15012
```

Comment:
Why not just fall-though to the one below? Is there a reason we wouldn't want `istiod-revision.istio-system.svc`?

Reply:
when we are in remote cluster,  we will create `istiod` or `istiod-remote` endpoint.  gateways will get `discoveryAddress` from the def here.

Reply:
Right, but we could have `if and (.Values.global.remotePilotAddress .Values.pilot.enabled` resulting in istiod-remote, because we already have a branch that gives `discoveryAddress: istiod`.

If we're doing this in a separate branch because we don't want to include the revision intentionally, can we call that out with a comment?

Reply:
yes, This is to align the endpoint def in https://github.com/istio/istio/blob/master/manifests/charts/base/templates/endpoints.yaml#L2

you see we actually will create `istiod` or `istiod-remote`